### PR TITLE
Plugin support

### PR DESCRIPTION
**Pull request by vladimanaev**
Added extension utils
Added even traffic allocation mode to campaign
Added cpa, cvr, actions columns to top campaign content report

### CODE REVIEW

File: src/main/java/com/taboola/backstage/Backstage.java [hunk 1]
```diff
@@ -176,15 +186,17 @@ public Backstage build() {
             organizeState();
             CommunicationConfig config = new CommunicationConfig(baseUrl, authBaseUrl, connectionTimeoutMillis, readTimeoutMillis, writeTimeoutMillis, userAgent, debug);
             CommunicationFactory communicator = new CommunicationFactory(config);
+            BackstageEndpointsFactory endpointsFactory = new BackstageEndpointsFactoryImpl(communicator);
```
Comment (barkamir) [inline src/main/java/com/taboola/backstage/Backstage.java]: I wasn't able to understand how one will be able to give different implementation to an endpoint.
in other words, how we will use it in our GW to give different impl ?

File: src/main/java/com/taboola/backstage/internal/extensions/BackstageExtensionUtilsImpl.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.taboola.backstage.internal.extensions;
+
+/**
+ * Created by vladi
+ * Date: 3/4/2018
+ * Time: 12:37 AM
+ * By Taboola
+ */
+public class BackstageExtensionUtilsImpl implements BackstageExtensionUtils {
```
Comment (barkamir) [inline src/main/java/com/taboola/backstage/internal/extensions/BackstageExtensionUtilsImpl.java]: maybe I miss something - why do we need to create this abstraction and not use directly the EndpointFactory ?

Comment (barkamir) [inline src/main/java/com/taboola/backstage/internal/extensions/BackstageExtensionUtilsImpl.java]: why is it utils ? currently it has only one method which return the endpoint factory, clearly not "utils".
I suggest to rename it to EndpointFactoryProvider/Creator or such... if there is a need to add another method to here in the future - we either will change the name of the class to be more global (but not utils), or add it to a new class.

Comment (vladimanaev) [reply src/main/java/com/taboola/backstage/internal/extensions/BackstageExtensionUtilsImpl.java]: @barkamir was thinking of backward compatibility in future releases

Comment (vladimanaev) [reply src/main/java/com/taboola/backstage/internal/extensions/BackstageExtensionUtilsImpl.java]: @barkamir I dont want to change often interfaces to be more backward compatible (as much as possible). But surely we can rename it to something more clearer.

**Pull request merged**